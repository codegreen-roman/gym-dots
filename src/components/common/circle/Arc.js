import React from 'react'
import { string } from 'prop-types'
import { Path } from './Arc.glamor'

export const Arc = ({ d, arcClass }) => {
    return <Path d={d} arcClass={arcClass} />
}

Arc.propTypes = {
    d: string.isRequired,
    arcClass: string.isRequired
}
