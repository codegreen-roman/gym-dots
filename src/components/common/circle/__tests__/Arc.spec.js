import React from 'react'
import { render } from 'enzyme'
import { Arc } from '../Arc'

const defaultProps = {
    d: 'M 205.1056516 79.0983006 A 100 100 0 0 0 127.3648178 11.5192247',
    arcClass: 'toBeDone'
}

const setup = (arcClass=defaultProps.arcClass, d = defaultProps.d) => {
    return render(<Arc d={d} arcClass={arcClass} />)
}

describe('Arc Component', () => {
    let wrapper
    let path

    beforeEach(() => {
        wrapper = setup()
        path = 'M 110 10 A 100 100 0 0 0 21.7052407 63.0528437'
    })

    test('should render correct color by default', () => {
        expect(wrapper).toMatchSnapshotWithGlamor()
    })

    test('should render correct color when completed', () => {
        wrapper = setup('completed')
        expect(wrapper).toMatchSnapshotWithGlamor()
    })

    test('should render correct color when failed', () => {
        wrapper = setup('failed')
        expect(wrapper).toMatchSnapshotWithGlamor()
    })

    test('should apply path when provided', () => {
        wrapper = setup('toBeDone', 'M 110 10 A 100 100 0 0 0 21.7052407 63.0528437')
        expect(wrapper).toMatchSnapshot()
    })

})
