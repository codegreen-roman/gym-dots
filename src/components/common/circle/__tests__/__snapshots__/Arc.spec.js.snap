// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arc Component should render correct color by default 1`] = `
.css-8plr4h,
[data-css-8plr4h] {
  fill: none;
  stroke-width: 12px;
  stroke: lightgrey;
}

<path
  class="css-8plr4h"
  d="M 205.1056516 79.0983006 A 100 100 0 0 0 127.3648178 11.5192247"
/>
`;

exports[`Arc Component should render correct color when completed 1`] = `
.css-zw8vjp,
[data-css-zw8vjp] {
  fill: none;
  stroke-width: 12px;
  stroke: lime;
}

<path
  class="css-zw8vjp"
  d="M 205.1056516 79.0983006 A 100 100 0 0 0 127.3648178 11.5192247"
/>
`;

exports[`Arc Component should render correct color when failed 1`] = `
.css-1tskx0m,
[data-css-1tskx0m] {
  fill: none;
  stroke-width: 12px;
  stroke: tomato;
}

<path
  class="css-1tskx0m"
  d="M 205.1056516 79.0983006 A 100 100 0 0 0 127.3648178 11.5192247"
/>
`;
